pkg/recordset/recordset_test.go

```diff
@@ -1,29 +1,790 @@
+		},
+		{
+			name: "case 3: do not create already existing stack",
+			sourceStacks: []cloudformation.Stack{
+				cloudformation.Stack{
+					StackName:   aws.String("cluster-foo-guest-main"),
+					StackStatus: aws.String(cloudformation.StackStatusCreateComplete),
+				},
+			},
+			targetStacks: []cloudformation.Stack{
+				cloudformation.Stack{
+					StackName:   aws.String("cluster-foo-guest-recordsets"),
+					StackStatus: aws.String(cloudformation.StackStatusCreateComplete),
+				},
+			},
+			expectedCreatedStacks: nil,
+		},
+		{
+			name:         "case 4: do not create stack when there is no source",
+			sourceStacks: nil,
+			targetStacks: []cloudformation.Stack{
+				cloudformation.Stack{
+					StackName:   aws.String("cluster-foo-guest-recordsets"),
+					StackStatus: aws.String(cloudformation.StackStatusCreateComplete),
+				},
+			},
+			expectedCreatedStacks: nil,
+		},
+		{
+			name:                  "case 5: do not create stack when there is no source and target",
+			sourceStacks:          nil,
+			targetStacks:          nil,
+			expectedCreatedStacks: nil,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			logger, err := micrologger.New(micrologger.Config{IOWriter: ioutil.Discard})
+			if err != nil {
+				t.Fatalf("micrologger.New: %v", err)
+			}
+
+			sourceClient := newSourceWithStacks(tc.sourceStacks)
+			targetClient := newTargetWithStacks(tc.targetStacks)
+
+			c := &Config{
+				Logger:               logger,
+				Installation:         installation,
+				SourceClient:         sourceClient,
+				TargetClient:         targetClient,
+				TargetHostedZoneID:   zoneID,
+				TargetHostedZoneName: zoneName,
+			}
+			m, err := NewManager(c)
+			if err != nil {
+				t.Fatalf("NewManager: %v", err)
+			}
+
+			err = m.createMissingTargetStacks(tc.sourceStacks, tc.targetStacks)
+			if err != nil {
+				t.Fatalf("m.Sync: %v", err)
+			}
+
+			sort.Strings(targetClient.createdStacks)
+			sort.Strings(tc.expectedCreatedStacks)
+
+			if !reflect.DeepEqual(tc.expectedCreatedStacks, targetClient.createdStacks) {
+				t.Errorf("created, expected %v  got %v", tc.expectedCreatedStacks, targetClient.createdStacks)
+			}
+		})
+	}
+}
+
+// TestCreateMissingStacks_Statuses tests Manager.createMissingTargetStacks.
+//
+// Creation is only allowed when source stack has status *_COMPLETE except DELETE_COMPLETE.
+func TestCreateMissingStacks_Statuses(t *testing.T) {
+	var (
+		installation = "installation"
+		zoneID       = "zoneID"
+		zoneName     = "zoneName"
+	)
+
+	tcs := []struct {
+		name         string
+		status       string
+		expectCreate bool
+	}{
+		{
+			name:         "case 0: create stack when source status is create complete",
+			status:       cloudformation.StackStatusCreateComplete,
+			expectCreate: true,
+		},
+		{
+			name:         "case 1: create stack when source status is update complete",
+			status:       cloudformation.StackStatusUpdateComplete,
+			expectCreate: true,
+		},
```

Comment:
So we basically create target stack when source is CreateComplete of UpdateComplete? That makes sense to me.

Reply:
Yes that's what we do